libraries/WiFiClientSecure/src/ssl_client.cpp

```diff
@@ -313,10 +315,20 @@ void stop_ssl_socket(sslclient_context *ssl_client, const char *rootCABuff, cons
         ssl_client->socket = -1;
     }
 
+    // avoid memory leak if ssl connection attempt failed
+    if (ssl_client->ssl_conf.ca_chain != NULL) {
```

Comment:
Testing for NULL is superficial and considered bad taste by some when the pointer is only handed over to free() anyway. The pointers below are not tested and thus this introduces a confusing inconstancy. IMHO.